artifact-size-enforcer/src/main/java/org/codice/ddf/admin/beta/enforcer/ArtifactSizeEnforcerRule.java

```diff
@@ -0,0 +1,202 @@
+import org.codehaus.plexus.component.configurator.expression.ExpressionEvaluationException;
+import org.codehaus.plexus.util.StringUtils;
+
+/**
+ * When the ArtifactSizeEnforceRule is active, it will look up the packaging type and enforce a maximum artifact size.
+ * If the packing type is unknown to the rule, enforcement will be skipped. The only currently supported packing types are bundle and jar.
+ *
+ * The following arguments can be specified to the ArtifactSizeEnforcerRule:
+ *
+ * maxArtifactSize - If the specified artifact is larger than this maxArtifactSize then an exception will be thrown. The unit of this argument should end with `_B` (Bytes), `_KB` (Kilobytes), or `_MB` (MegaBytes). By default, maxArtifactSize is set to 1_MB.
+ * artifactLocation - A path to the file to be checked. If a artifactLocation is not looked up, a default artifact will attempt to be found using the project version, packaging and artifactId.
+ * skip - The ArtifactSizeEnforceRule will not run.
+ */
+public class ArtifactSizeEnforcerRule implements EnforcerRule {
+
+    public static final String PROJECT_PACKAGING_PROP = "${project.packaging}";
+    public static final String PROJECT_ARTIFACT_ID_PROP = "${project.artifactId}";
+    public static final String PROJECT_VERSION_PROP = "${project.version}";
+    public static final String PROJECT_BUILD_DIR_PROP = "${project.build.directory}";
+
+    public static final String DEFAULT_MAX_ARTIFACT_SIZE = "1_MB";
+    public static final String BYTES = "_B";
+    public static final String MEGA_BYTES = "_MB";
+    public static final String KILO_BYTES = "_KB";
+
+    public static final String JAR = "jar";
+    public static final String BUNDLE = "bundle";
+    public static final List<String> SUPPORTED_PACKAGE_TYPES = Arrays.asList(JAR, BUNDLE);
+
+    //Rule arguments
+    private String maxArtifactSize;
+    private String artifactLocation;
+    private boolean skip;
+
+    private EnforcerRuleHelper helper;
+    private Log logger;
+
+    @Override
+    public void execute(EnforcerRuleHelper helper) throws EnforcerRuleException {
+        this.helper = helper;
+        this.logger = helper.getLog();
+
+        if (skip) {
+            return;
+        }
+
+        String packaging = getPackaging();
+        if (!SUPPORTED_PACKAGE_TYPES.contains(packaging)) {
+            logger.info(String.format(
+                    "Unsupported package type %s. Skipping artifact size enforcement.",
+                    packaging));
+            return;
+        }
+
+        String artifactPath = getArtifactPath();
+        long maxArtifactSizeBytes = maxArtifactSizeToBytes();
+        long artifactSize = getArtifact(artifactPath).length();
+
+        if (artifactSize >= maxArtifactSizeBytes) {
+            throw new EnforcerRuleException(String.format(
+                    "The specified artifact is larger than the set maximum artifact size. \n\n\tArtifact: %s\n\tArtifact Size: %s Bytes\n\tMax Artifact Size: %s Bytes\n"
```

Comment:
❗️ Use `%n` in calls to `String.format`.

Don't concatenate inside a `String.format` call.

Save your format String off as a const; it'd be perfectly reasonable to concat there if line length is an issue. Still use `%n` for newline, however.

Also, you're passing in `long`s, not `String`s; change the `%s` references to `%d` accordingly.